experimental/builder/test/unit_conv_tensor_layout.cpp

```diff
@@ -9,249 +9,257 @@
 
 namespace {
 
-namespace ckb = ::ck_tile::builder;
-using ::ck_tile::builder::DataType;
-using ::ck_tile::builder::ElementwiseOperation;
-using ::ck_tile::builder::TensorLayout;
-using ::ck_tile::builder::factory::internal::AuxiliaryTensorLayouts;
-using ::ck_tile::builder::factory::internal::ConvTensorLayouts;
-using ::ck_tile::builder::factory::internal::LayoutToCK;
-
-using namespace ::ck_tile::builder::test;
-using enum ::ck_tile::builder::ConvDirection;
+namespace ckb = ck_tile::builder;
+using ck_tile::builder::DataType;
+using ck_tile::builder::ElementwiseOperation;
+using ck_tile::builder::TensorLayout;
+using ck_tile::builder::factory::internal::AuxiliaryTensorLayouts;
+using ck_tile::builder::factory::internal::ConvTensorLayouts;
+using ck_tile::builder::factory::internal::LayoutToCK;
+using ck_tile::builder::test::ConvolutionTensor;
+using ck_tile::builder::test::ConvSignature;
+using ck_tile::builder::test::TensorConfig;
+using ck_tile::builder::test::TensorOperation;
+using ::testing::StaticAssertTypeEq;
+
+namespace enums {
+using enum ck_tile::builder::ConvDirection;
+using enum ck_tile::builder::TensorLayout;
+using enum ck_tile::builder::DataType;
+} // namespace enums
 
 TEST(ConvTensorLayout, AssignsLayoutsFor1D_NWGC_GKXC_NWGK)
 {
-    static constexpr auto sig =
-        ConvSignature<>{.spatial_dim            = 1,
-                        .direction              = FORWARD,
-                        .data_type              = DataType::FP16,
-                        .accumulation_data_type = DataType::FP32,
-                        .input                  = {.config = {.layout = TensorLayout::NWGC}},
-                        .weight                 = {.config = {.layout = TensorLayout::GKXC}},
-                        .output                 = {.config = {.layout = TensorLayout::NWGK}}};
+    using namespace enums;
+    static constexpr auto sig = ConvSignature<>{.spatial_dim            = 1,
+                                                .direction              = FORWARD,
+                                                .data_type              = FP16,
+                                                .accumulation_data_type = FP32,
+                                                .input  = {.config = {.layout = NWGC}},
+                                                .weight = {.config = {.layout = GKXC}},
+                                                .output = {.config = {.layout = NWGK}}};
 
     using TensorLayouts = ConvTensorLayouts<sig, 1, FORWARD>;
 
-    EXPECT_TRUE((std::is_same_v<TensorLayouts::ALayout, ck::tensor_layout::convolution::NWGC>));
-    EXPECT_TRUE((std::is_same_v<TensorLayouts::BLayout, ck::tensor_layout::convolution::GKXC>));
-    EXPECT_TRUE((std::is_same_v<TensorLayouts::ELayout, ck::tensor_layout::convolution::NWGK>));
-    EXPECT_TRUE((std::is_same_v<TensorLayouts::DsLayout, ck::Tuple<>>));
+    StaticAssertTypeEq<TensorLayouts::ALayout, ck::tensor_layout::convolution::NWGC>();
```

Comment:
This was bad advice I gave you. I tried this out, and it gives a compiler error instead of a failing test, which makes the test much more difficult to identify and fix. I thought I shared that in chat, but I clearly did not communicate that to you. We probably don't want to use it here, but you can try making a failing test and see what you think is best. I'm sorry for my confusion; it wasn't clear this was a bad suggestion until I broke a test.